crates/pallet-domains/src/nominator_position.rs

```diff
@@ -0,0 +1,221 @@
+    use crate::bundle_storage_fund;
+
+    let storage_fund_redeem_price = bundle_storage_fund::storage_fund_redeem_price::<T>(
+        operator_id,
+        operator_total_storage_fee,
+    );
+
+    storage_fund_redeem_price.redeem(nominator_storage_fee)
+}
+
+/// Processes pending withdrawals for the nominator
+fn process_withdrawals<T: Config>(
+    operator_id: OperatorId,
+    nominator_account: &T::AccountId,
+    current_share_price: &crate::staking::SharePrice,
+) -> Vec<sp_domains::PendingWithdrawal<BalanceOf<T>, DomainBlockNumberFor<T>>> {
+    let Some(withdrawal) = Withdrawals::<T>::get(operator_id, nominator_account) else {
+        return Vec::new();
+    };
+
+    let mut pending_withdrawals = Vec::with_capacity(
+        withdrawal.withdrawals.len()
+            + if withdrawal.withdrawal_in_shares.is_some() {
+                1
+            } else {
+                0
+            },
+    );
+
+    // Process regular withdrawals
+    pending_withdrawals.extend(withdrawal.withdrawals.into_iter().map(|w| {
+        sp_domains::PendingWithdrawal {
+            amount: w.amount_to_unlock,
+            unlock_at_block: w.unlock_at_confirmed_domain_block_number,
+        }
+    }));
+
+    // Process withdrawal in shares
+    if let Some(withdrawal_in_shares) = withdrawal.withdrawal_in_shares {
+        let withdrawal_amount =
+            OperatorEpochSharePrice::<T>::get(operator_id, withdrawal_in_shares.domain_epoch)
+                .map(|epoch_share_price| {
+                    epoch_share_price.shares_to_stake::<T>(withdrawal_in_shares.shares)
+                })
+                .unwrap_or_else(|| {
+                    current_share_price.shares_to_stake::<T>(withdrawal_in_shares.shares)
+                });
+
+        pending_withdrawals.push(sp_domains::PendingWithdrawal {
+            amount: withdrawal_amount,
+            unlock_at_block: withdrawal_in_shares.unlock_at_confirmed_domain_block_number,
+        });
+    }
+
+    pending_withdrawals
+}
+
+/// Returns the complete nominator position for a given operator and account.
+///
+/// This calculates the total position including:
+/// - Current stake value (converted from shares using instant share price including rewards)
+/// - Total storage fee deposits (known + pending)
+/// - Pending deposits (not yet converted to shares)
+/// - Pending withdrawals (with unlock timing)
+///
+/// Returns None if no position exists for the given nominator and operator.
+pub fn nominator_position<T: Config>(
+    operator_id: OperatorId,
+    nominator_account: T::AccountId,
+) -> Option<sp_domains::NominatorPosition<BalanceOf<T>, DomainBlockNumberFor<T>>> {
+    use sp_domains::NominatorPosition;
+
+    // Fetch core data needed for position calculation
+    let position_data = fetch_position_data::<T>(operator_id, &nominator_account)?;
+
+    // Calculate current shares and storage fees from deposits
+    let (total_shares, total_storage_fee_deposit, pending_deposits) =
```

Comment:
It may also be helpful to include `total_shares` in the runtime API (or add a new runtime API for that), since now we only support withdrawal by share, and there is no efficient way to query the total share that can be withdrawn.

Reply:
Done